src/RscChart.tsx

```diff
@@ -95,6 +95,8 @@ export const RscChart = forwardRef<ChartHandle, RscChartProps>(
 		forwardedRef
 	) => {
 		// uuid is used to make a unique id so there aren't duplicate ids if there is more than one Chart component in the document
+		const doAnimateFromZero = useRef(true);
```

Comment:
Correct me if this is wrong, but I think this value should be state instead?  The boolean value probably affects the render result so it can be counterintuitive to keep it as a ref.

You probably already know this, but React processes state changes in a batch.  I don't know how it interops with refs though